### PR TITLE
Fix parameters in PPS RawToDigiConverter, and clean up a few related configs

### PR DESCRIPTION
**Pull request by perrotta**
#### PR description:

(This replaces #41784 that I somehow messed up)

Fix a few parameters in PPS RawToDigiConverter:

- Make "useOlderT2TestFile" tracked
- Make "useOlderT2TestFile", "printErrorSummary", "printUnknownFrameSummary" bool parameters, instead of unsigned int
- Propagate those modified parameters in the corresponding python configs

While doing so I also:

- Made constant a few class members of RawToDigiConverter that should have been so
- Moved to a safer syntax, with no explicit type declaration, in the touched configs when possible

This is a follow up of the https://github.com/cms-sw/cmssw/pull/41777#issuecomment-1564189810 posted when merging PR https://github.com/cms-sw/cmssw/pull/41777

#### PR validation:

It builds and a few matrix workflows run

@oljemark @forthommel and https://github.com/orgs/cms-sw/teams/ctpps-dpg-l2 please check that this actually corresponds to what you need (nothing should change in the final results with respect to the code currently merged in the master release). I did not try the totemt2_dqm_test because I miss some input: could you please give it a try and report if you get any error from it?

### CODE REVIEW

**Comment by oljemark** [comment]
Hello, @perrotta . Yesterday I was able to run the fixes you requested (without the code improvements you introduce here) in branch CTPPS/cmssw:fo_nt2_mapfix , which also contained some bugfixes to be able to read newly produced T2 test files from this Friday, so now I extracted only part you asked for to branch https://github.com/cms-sw/cmssw/compare/master...CTPPS:cmssw:fo_nt2_bugfix2 , for which at least the New T2 unit test in RecoPPS/Local runs successfully. I'll try out your code next.

File: EventFilter/CTPPSRawToDigi/python/totemTimingRawToDigi_cfi.py
Comment (oljemark) [inline EventFilter/CTPPSRawToDigi/python/totemTimingRawToDigi_cfi.py:1]: I see some noncapitalized "false" here too.

Comment (perrotta) [reply EventFilter/CTPPSRawToDigi/python/totemTimingRawToDigi_cfi.py:1]: Thank you @oljemark , fixed now

**Comment by perrotta** [comment]
> Hello, @perrotta . Yesterday I was able to run the fixes you requested (without the code improvements you introduce here) in branch CTPPS/cmssw:fo_nt2_mapfix , which also contained some bugfixes to be able to read newly produced T2 test files from this Friday, so now I extracted only part you asked for to branch [master...CTPPS:cmssw:fo_nt2_bugfix2](https://github.com/cms-sw/cmssw/compare/master...CTPPS:cmssw:fo_nt2_bugfix2) , for which at least the New T2 unit test in RecoPPS/Local runs successfully. I'll try out your code next.

Please fill free to take whatever you want from this PR and use with your own: the "official" @cms-sw/ctpps-dpg-l2 code will supersede this fix when ready, if you think so

File: EventFilter/CTPPSRawToDigi/python/totemVFATRawToDigi_cfi.py [hunk 1]
```diff
@@ -38,11 +38,14 @@
...
     EC_fraction = cms.untracked.double(0.6),
     BC_fraction = cms.untracked.double(0.6),
+
+    # True is only used for special test runs
+    useOlderT2TestFile = cms.bool(false),
     
-    # if non-zero, prints a per-VFAT error summary at the end of the job
-    printErrorSummary = cms.untracked.uint32(0),
+    # prints a per-VFAT error summary at the end of the job
+    printErrorSummary = cms.untracked.bool(false),
```
Comment (oljemark) [inline EventFilter/CTPPSRawToDigi/python/totemVFATRawToDigi_cfi.py:46]: I see some here too.

**Comment by perrotta** [comment]
please test

**Comment by oljemark** [comment]
Hi, @perrotta . I added a fix for the New T2 unit test in RecoPPS/Local here: https://github.com/cms-sw/cmssw/compare/master...CTPPS:cmssw:fo_fixParamInCtppsRawToDigiConverter , after which the unit test succeeded with your code just as in cmssw/master.

**Comment by perrotta** [comment]
> * You potentially added 13915 lines to the logs

Trackiness was forgotten for existsAs in RawToDigiConverter.cc: fixed with the latest commit

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
@oljemark I've added your extra commit here (thank you again).
If you are fine with this PR we can go ahead with it: otherwise, please open a new one and I will close this one.

**Comment by oljemark** [comment]
Hi, @perrotta . When I did cms-checkdeps for my own branch fo_nt2_mapfix2 and now yours too I got several unit test failures in DQM/Integration and one in CalibPPS/TimingCalibration , but when I tried in the from-CMSSW_13_2_X_2023-05-28-2300 branch I also got these failures, so it doesn't seem to come from your code, most being unreachable ROOT files?

[testing-DQMIntegration-perrotta-2023maj28-2300-nightly.log](https://github.com/cms-sw/cmssw/files/11590363/testing-DQMIntegration-perrotta-2023maj28-2300-nightly.log)
[testing-DQMIntegration-nightlyMaster-2023maj28-2300-nightly.log](https://github.com/cms-sw/cmssw/files/11590364/testing-DQMIntegration-nightlyMaster-2023maj28-2300-nightly.log)
[testing2-CalibPPSTimingCalibration-nightlyMaster.txt](https://github.com/cms-sw/cmssw/files/11590412/testing2-CalibPPSTimingCalibration-nightlyMaster.txt)

Your branch:
Pass    5s ... CalibPPS/TimingCalibration/testDiamondLUTCalibration
Fail   74s ... CalibPPS/TimingCalibration/testHptdcPcl
Fail    5s ... DQM/Integration/TestDQMOnlineClient-beam_dqm_sourceclient
Pass    8s ... DQM/Integration/TestDQMOnlineClient-beamhlt_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-beampixel_dqm_sourceclient
Fail   12s ... DQM/Integration/TestDQMOnlineClient-csc_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-ctpps_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-dt4ml_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-dt_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-ecal_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-es_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-fed_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-gem_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-hcal_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-hcalreco_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-hlt_dqm_clientPB
Fail    5s ... DQM/Integration/TestDQMOnlineClient-hlt_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-info_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-l1tstage2_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-l1tstage2emulator_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-mutracking_dqm_sourceclient
Pass   10s ... DQM/Integration/TestDQMOnlineClient-onlinebeammonitor_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-pixel_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-pixellumi_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-rpc_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-scal_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-sistrip_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-visualization
Fail    5s ... DQM/Integration/TestDQMOnlineClient-visualization_secondInstance

On latest nightly branch: 
Pass    3s ... CalibPPS/TimingCalibration/testDiamondLUTCalibration
Fail   42s ... CalibPPS/TimingCalibration/testHptdcPcl
Fail    7s ... DQM/Integration/TestDQMOnlineClient-beam_dqm_sourceclient
Pass    7s ... DQM/Integration/TestDQMOnlineClient-beamhlt_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-beampixel_dqm_sourceclient
Fail   11s ... DQM/Integration/TestDQMOnlineClient-csc_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-ctpps_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-dt4ml_dqm_sourceclient
Fail    4s ... DQM/Integration/TestDQMOnlineClient-dt_dqm_sourceclient
Fail    4s ... DQM/Integration/TestDQMOnlineClient-ecal_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-es_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-fed_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-gem_dqm_sourceclient
Fail    6s ... DQM/Integration/TestDQMOnlineClient-hcal_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-hcalreco_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-hlt_dqm_clientPB
Fail    5s ... DQM/Integration/TestDQMOnlineClient-hlt_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-info_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-l1tstage2_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-l1tstage2emulator_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-mutracking_dqm_sourceclient
Pass    8s ... DQM/Integration/TestDQMOnlineClient-onlinebeammonitor_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-pixel_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-pixellumi_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-rpc_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-scal_dqm_sourceclient
Fail    5s ... DQM/Integration/TestDQMOnlineClient-sistrip_dqm_sourceclient
Fail    4s ... DQM/Integration/TestDQMOnlineClient-visualization
Fail    5s ... DQM/Integration/TestDQMOnlineClient-visualization_secondInstance

**Comment by oljemark** [comment]
With the same packages checked out in nightly, that is.

**Comment by perrotta** [comment]
please test
(Let try testing here)

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
hold

**Comment by perrotta** [comment]
@oljemark this is now fully signed. Please let us know if it can be merged as such, or if you have another PR ready to be submitted that can supersede this one

**Comment by grzanka** [comment]
> Hi, @perrotta . When I did cms-checkdeps for my own branch fo_nt2_mapfix2 and now yours too I got several unit test failures in DQM/Integration and one in CalibPPS/TimingCalibration , but when I tried in the from-CMSSW_13_2_X_2023-05-28-2300 branch I also got these failures, so it doesn't seem to come from your code, most being unreachable ROOT files?

I think that is an unreachable ROOT file. When running `CalibPPS/TimingCalibration/testHptdcPcl` test on lxplus, cmsRun tries to access `/store/data/Run2022B/AlCaPPS/RAW/v1/000/355/207/00000/c23440f4-49c0-44aa-b8f6-f40598fb4705.root` which is on TAPE. The tests fails.

On the other hand, the cmsbot on Jenkins tries to access that file via `/eos/cms/store/user/cmsbuild/store/data/Run2022B/AlCaPPS/RAW/v1/000/355/207/00000/c23440f4-49c0-44aa-b8f6-f40598fb4705.root `  (note `/eos/cms/store/user/cmsbuild/` in front) which is available on EOS and test passes.

**Comment by oljemark** [comment]
Hi, @perrotta . I've gotten several DQM plot requests and found some additional functionality was needed in the DataFormats/TotemReco, so I'm going with the PR still in CTPPS, #75, with some work still to be done to get it ready by tomorrow.

**Comment by perrotta** [comment]
Already included in (and therefore superseded by) #41859

**Pull request closed without merging**